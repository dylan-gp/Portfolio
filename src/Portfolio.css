.portfolio-img {
  height: 24vw;
  width: 40vw;
  margin: auto 4vw;
}



.portfolio-img:hover {
  cursor: pointer;
}

.port-container {
  width: 60vw;
  overflow: hidden;
  margin: auto;
  position: relative;
  z-index: 100;
  min-height: 100vh;
  vertical-align: middle;
  padding-top: 4em;
}

.portfolio-list {
  padding: 0;
  padding-top: 4em;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin: auto;
  align-items: center;
  list-style: none;
  overflow-x: scroll;
  width: 48vw;
}

.portfolio-list::-webkit-scrollbar {
  display: none;
}


.scroll-arrow-right {
  position: absolute;
  right: 4%;
  height: 8vw;
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  -o-transition: 0.4s;
  transition: 0.4s; 
  opacity: .4;
}
.scroll-arrow-right-hide {
  position: absolute;
  right: 4%;
  height: 8vw;
  display: none;
}
.scroll-arrow-right:hover {
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  -o-transition: 0.4s;
  transition: 0.4s;
  opacity: 1;
  cursor: pointer;
}
.scroll-arrow-left{
  position: absolute;
  left: 4%;
  height: 8vw;
  transform: rotate(180deg);
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  -o-transition: 0.4s;
  transition: 0.4s;
  opacity: .4;
}
.scroll-arrow-left-hide {
  position: absolute;
  left: 4%;
  height: 8vw;
  display: none;
}
.scroll-arrow-left:hover {
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  -o-transition: 0.4s;
  transition: 0.4s;
  opacity: 1;
  cursor: pointer;
}

.piece-name {
  text-align: center;
}