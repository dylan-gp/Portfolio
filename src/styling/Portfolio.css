.portfolio-item-img  {
  height: 15vw;
  margin: auto;
  vertical-align: middle;
  transition: height .5s;
}
.portfolio-item-img:hover {
  height: 25%;
}
.portfolio-item-img-container {
  vertical-align: middle;
  display: flex;
  justify-content: center;
  background-color: rgb(67, 32, 160);
  border: outset 2px black;
  border-radius: 2%;
  height: 35vw;
  width: 30vw;
  overflow: hidden;
}
.portfolio-item-overlay:hover + 
.portfolio-item-img-container 
.portfolio-item-img {
  height: 20vw;
  transition: height .5s;
}
.portfolio-item-overlay {
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  position: absolute;
  height: 35vw;
  width: 30vw;
  z-index: 1;
  border: outset 2px whitesmoke;
  border-radius: 2%;
  opacity: 1;
  visibility: visible;
  transition: opacity 0.5s, visibility 0.5s;
}
.portfolio-item-overlay:hover {
  cursor: pointer;
  opacity: 0;
  visibility: none;
  transition: opacity 0.5s, visibility 0.5s;
}
.portfolio-overlay-img {
  max-width: 10vw;
  max-height: 8vw;
}
.portfolio-item-img:hover {
  cursor: pointer;
}
.portfolio-item-list {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  vertical-align: middle;
  padding: 0;
  margin: 2vw;
}
.port-container {
  width: 100vw;
  margin: auto 0;
  position: relative;
  overflow: hidden;
  z-index: 100;
  vertical-align: middle;
  padding: 0;
  padding-top: 4em;
  margin-bottom: 4em;
  min-height: 100vh;
}
.portfolio-list {
  box-sizing: border-box;
  padding: 0;
  width: 100vw;
}
@media screen and (max-width: 500px) {
  .port-container {
    width: 100vw;
    padding: 1em 0;
    min-height: 0;
  }
  .portfolio-item-list {
    width: 45vw;
    margin: 1vw;
  }
  .portfolio-item-overlay {
    width: 45vw;
  }
  .portfolio-item-img-container {
    width: 45vw;
  }
  .portfolio-list {
    padding: 0;
    margin: 0;
  }
}