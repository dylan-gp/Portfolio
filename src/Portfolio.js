import React from 'react';
import Piece from './Piece';
import CodersCodeImg from './CodersCode.png';
import CodersCodeClientImg from './CodersCodeClient.png';
import CodersCodeAPIImg from './CodersCodeAPI.png';
import ZoomToYoutube from './ZoomToYoutube.png';
import ZoomToYoutube2 from './ZoomToYoutube2.png';
import arrow from './arrow.svg';

import './Portfolio.css';


export default class Portfolio extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      codersCode: false,
      hideLeft: true,
      hideRight: false,
      currentText: '',
      size: window.matchMedia("screen and (max-width: 500px)").matches ? .80 : .48
    }
    this.scroll = this.scroll.bind(this);
    this.popUp = this.popUp.bind(this);
    this.popDown = this.popDown.bind(this);
    this.arrowVisibility = this.arrowVisibility.bind(this);
    this.resize = this.resize.bind(this);
    this.codersCodeText = "The Coders Code";
    this.notAvail = "N/A Yet"
    this.fillText = this.fillText.bind(this);
    this.stopText = this.stopText.bind(this);
  };

  componentDidMount() {
    this.list.addEventListener('scroll', this.arrowVisibility);
    window.addEventListener('resize', this.resize);
  }
  componentWillUnmount() {
    window.removeEventListener('resize', this.resize);
  }

  popUp(piece) {
    if (piece === 'coders') {
      this.setState({ codersCode: true });
      // document.body.classList.add('no-scroll');
    }
  }

  popDown(piece) {
    if (piece === 'coders') {
      this.setState({ codersCode: false });
      // document.body.classList.remove('no-scroll');
    }
  }

  resize() {
    const prev = this.state.size;
    this.setState({size: window.matchMedia("screen and (max-width: 500px)").matches ? .80 : .48});
    if (prev !== this.state.size) this.list.scrollLeft = 0;

  }
  arrowVisibility() {
    const width = window.innerWidth * this.state.size;
    if (this.list.scrollLeft === 0 && !this.state.hideLeft) this.setState({ hideLeft: true });
    if (this.list.scrollLeft > 0 && this.state.hideLeft) this.setState({ hideLeft: false });
    if (this.list.scrollLeft >= Math.floor(width * 2) && !this.state.hideRight) this.setState({ hideRight: true });
    if (this.list.scrollLeft < Math.floor(width * 2) && this.state.hideRight) this.setState({ hideRight: false });
  }
  scroll(boo) {
    const width = Math.round(window.innerWidth * this.state.size);
    boo ? this.list.scrollLeft += width : this.list.scrollLeft -= width;
    // this.interval = setInterval(() => boo ? this.list.scrollLeft += window.innerWidth * .32 : this.list.scrollLeft -= window.innerWidth * .32, 1);
  }

  fillText(text = '') {
    clearInterval(this.int);
    if (!this.i) this.i = 0;
    if (this.state.currentText.length === 0) clearInterval(this.int)
    this.int = setInterval(() => {
      this.setState({currentText: this.state.currentText + text[this.i]});
      this.i++;
      if (this.state.currentText.length === text.length) clearInterval(this.int);
    }, 20);
  }
  stopText() {
    clearInterval(this.int);
    this.int = setInterval(() => {
      if (this.state.currentText === '' || this.state.currentText === undefined || this.state.currentText === 'undefined') {
        clearInterval(this.int);
        this.setState({currentText: ''});
      }
      else {
        this.setState({currentText: this.state.currentText.slice(0, -1)});
        this.i--;
      }
    }, 20);
  }

  unscroll() {
    clearInterval(this.interval);
  }
  render() {
    return (
      <div className="port-container">
        <h1 className="about-title">dylansPortfolio</h1>
        <ul ref={(ul) => this.list = ul } className="portfolio-list">
          <img 
            className={this.state.hideLeft ? "scroll-arrow-left-hide" : "scroll-arrow-left"} 
            onClick={() => this.scroll(false)} src={arrow} 
            alt="go left"
          />
          <li id="coders-code-img">
            <img 
              className="portfolio-img"
              onClick={() => this.popUp('coders')} src={CodersCodeImg}
              onMouseOver={() => this.fillText(this.codersCodeText)}
              onMouseLeave={() => this.stopText()}
              alt="Coders Code"
            />
          </li>
          {this.state.codersCode ? 
            <Piece
              popDown={this.popDown}
              image1={CodersCodeImg}
              image2={CodersCodeClientImg}
              image3={CodersCodeAPIImg} 
              githubClient="https://github.com/dys2/CodersCode"
              githubAPI="https://github.com/dys2/CodersCodeAPI"
              pageLink="http://www.thecoderscode.com/"
            /> : ''}
          <li>
            <img className="portfolio-img" src={ZoomToYoutube} alt="Zoom To Youtube"
            onMouseOver={() => this.fillText(this.notAvail)}
            onMouseLeave={() => this.stopText()}
            />
          </li>
          <li>
            <img className="portfolio-img" src={ZoomToYoutube2} alt="Zoom To Youtube2"
            onMouseOver={() => this.fillText(this.notAvail)}
            onMouseLeave={() => this.stopText()}
            />
          </li>
          <img
            className={this.state.hideRight ? "scroll-arrow-right-hide" : "scroll-arrow-right"}
            onClick={() => this.scroll(true)} src={arrow}
            alt="go right"
            />
        </ul>
        <div className="piece-name-div">
        <h2 className="piece-name" ref={(h2) => this.ccText = h2 } >{this.state.currentText}</h2>
        </div>
      </div>
    )
  }
}