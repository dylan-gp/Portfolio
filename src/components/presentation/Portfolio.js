import React from 'react';
import codersCodeImg from '../../images/coders-code-laptop.png';
import eviluatorImg from '../../images/eviluator-laptop.png';
import codersCodeLogo from '../../images/tcc-logo.png';
import eviluatorLogo from '../../images/evil-logo.png';
import z2ytLogo from '../../images/z2yt-logo.png';
import zoomImg from '../../images/zoom-laptop.png';
import portfolioImg from '../../images/portfolio-laptop.png';
import portfolioLogo from '../../images/portfolio-logo.png';
import '../../styling/Portfolio.css';

export default (props) => (
  <div className="port-container" ref={props.setRef}>
    <h1 className="about-title">dylansPortfolio</h1>
    <ul ref={ul => this.list = ul } className="portfolio-list">
      {
        PortfolioItems
          .map(
            item => 
              <PortfolioListItem item={item} />
          )
      }
    </ul>
  </div>
);

const PortfolioListItem = (props) => (
  <li className="portfolio-item-list" id={props.item.name}>
    <a
      className="portfolio-item-overlay"
      href={props.item.link}
      target="_blank"
    >
      <img
        className="portfolio-overlay-img"
        src={props.item.logo}
        alt={`${props.item.name} logo`}
      />
    </a>
    <div className="portfolio-item-img-container" >
      <img
        className="portfolio-item-img"
        src={props.item.img}
        alt={props.item.name}
      />
    </div>
  </li>
);

const PortfolioItems = [
  {
    name: 'Coders Code',
    img: codersCodeImg,
    logo: codersCodeLogo,
    link: 'http://www.thecoderscode.com/'
  },
  {
    name: 'Eviluator',
    img: eviluatorImg,
    logo: eviluatorLogo,
    link: 'https://www.youtube.com/watch?v=PNu4itiHiSQ&t=3m48s'
  },
  {
    name: 'Zoom To Youtube',
    img: zoomImg,
    logo: z2ytLogo,
    link: 'https://github.com/LocalZoomToYouTube/z2ytScript'
  },
  {
    name: 'Portfolio',
    img: portfolioImg,
    logo: portfolioLogo,
    link: 'https://github.com/dys2/Portfolio'
  }
];