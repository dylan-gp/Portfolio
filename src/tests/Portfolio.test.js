import React from 'react';
import Portfolio from '../components/Portfolio';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

Object.defineProperty(window, "matchMedia", {
  value: jest.fn(() => { return { matches: true } })
});

const header = shallow(<Portfolio />);

test('Component should match the snapshot', () => {
  const component = renderer.create(<Portfolio />);
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});

test('Component mounts with correct state', () => {
  const state = header.state();
  const expectedState = {
    codersCode: false,
    hideLeft: true,
    hideRight: false,
    currentText: ''
  };
  expect(state).toMatchObject(expectedState);
});