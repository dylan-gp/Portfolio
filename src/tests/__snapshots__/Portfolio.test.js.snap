// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should match the snapshot 1`] = `
<div
  className="port-container"
>
  <h1
    className="about-title"
  >
    dylansPortfolio
  </h1>
  <ul
    className="portfolio-list"
  >
    <img
      alt="go left"
      className="scroll-arrow-left-hide"
      onClick={[Function]}
      src="arrow.svg"
    />
    <li
      id="coders-code-img"
    >
      <img
        alt="Coders Code"
        className="portfolio-img"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        src="CodersCode.png"
      />
    </li>
    
    <li>
      <img
        alt="Zoom To Youtube"
        className="portfolio-img"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        src="ZoomToYoutube.png"
      />
    </li>
    <li>
      <img
        alt="Zoom To Youtube2"
        className="portfolio-img"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        src="ZoomToYoutube2.png"
      />
    </li>
    <img
      alt="go right"
      className="scroll-arrow-right"
      onClick={[Function]}
      src="arrow.svg"
    />
  </ul>
  <div
    className="piece-name-div"
  >
    <h2
      className="piece-name"
    >
      
    </h2>
  </div>
</div>
`;
